dispatcher.py

```diff
@@ -86,6 +88,13 @@ class Dispatcher:
     ```
     """
     def __init__(self, task_groups, max_workers_cnt, randomize):
+        if platform == "darwin":
+            try:
+                # Python 3
+                multiprocessing.set_start_method('fork')
+            except AttributeError:
+                # Python 2
+                pass
```

Comment:
I tried 'spawn' start method on Linux and was hit by 'No such file or directory' from `LogOutputWatcher.process_result()`. Sure, the `test/var/log` directory that is created in the costructor was removed. I found that it is because of reevaluation of the `lib/__init__.py` content in the child process (it removes `test/var`).

So 'spawn' causes re-execution of code in child processes and it is not easy (at least for me, at least now) to say what else will be executed in each process in addition to execution in the main process. We don't expect such behaviour: the code it not ready for that.

So I vote to keep 'fork' everywhere.

Reply:
no action is required

Reply:
In fact, it is the primary reason of choosing 'fork', not because we need to eliminate one lambda function. I would leave several words regarding this somewhere.

Reply:
I'm going to change the comment this way:

```diff
diff --git a/test-run.py b/test-run.py
index ce3558d..63392ff 100755
--- a/test-run.py
+++ b/test-run.py
@@ -219,11 +219,19 @@ def main_consistent():
 
 if __name__ == "__main__":
     # In Python 3 start method 'spawn' in multiprocessing module becomes
-    # default on Mac OS [1]. However 'spawn' method requires object serialization
-    # that doesn't work when objects use lambdas, whose for example used in
-    # class TestSuite (lib/test_suite.py). So this method is not acceptable
-    # for using with test-run. To workaround it can be possible to force
-    # fork mode with multiprocessing.set_start_method('fork').
+    # default on Mac OS [1].
+    #
+    # The 'spawn' method causes re-execution of some code, which is already
+    # executed in the main process. At least it is seen on the
+    # lib/__init__.py code, which removes the 'var' directory. Some other
+    # code may have side effects too, it requires investigation.
+    #
+    # The method also requires object serialization that doesn't work when
+    # objects use lambdas, whose for example used in class TestSuite
+    # (lib/test_suite.py).
+    #
+    # The latter problem is easy to fix, but the former looks more
+    # fundamental. So we stick to the 'fork' method now.
     if PY3:
         multiprocessing.set_start_method('fork')
```

I'll do the same change for the commit message. I hope you don't mind.

Reply:
And, while I'm here, the tiny change for consistency:

```diff
diff --git a/test-run.py b/test-run.py
index c2af31e..f820272 100755
--- a/test-run.py
+++ b/test-run.py
@@ -55,7 +55,8 @@ import time
 
 from lib import Options
 from lib.colorer import color_stdout
-from lib.utils import print_tail_n, PY3
+from lib.utils import print_tail_n
+from lib.utils import PY3
 from lib.worker import get_task_groups
 from lib.worker import get_reproduce_file
 from lib.worker import reproduce_task_groups
```

Reply:
Pushed.